project.sh

```diff
@@ -559,22 +559,34 @@ local-clean(){
 	esac
 }
 
+trim(){
+	echo "$@" | xargs
+}
+
+## args: [-r|--refresh|--watch] [-ff|--fail-fast] <focus>
+## Runs the Clojure unit tests using Kaocha
+## [-r|--refresh|--watch] Watches tests and source files for changes, and subsequently re-evaluates
+## [-ff|--fail-fast] Stop tests as soon as a single failure or error has occurred
+## <focus> Suite/namespace/var to focus on
 -test-clj () {
 	allow-snapshots
 	local cmd
-	#TODO make args order independent
-	case $1 in
-		-r|--refresh|--watch)
-			cmd='--watch'
-			shift;;
-	esac
-	case $1 in
-		-ff|--fail-fast)
-			cmd="$cmd --fail-fast"
-			shift;;
-	esac
-	if [ -n "$1" ];then
-		cmd="$cmd --focus $*"
+	local remaining
+	while [ -n "$1" ];do
+		case "$1" in
+			-r|--refresh|--watch)
+				cmd="$cmd --watch";;
+			-ff|--fail-fast)
+				cmd="$cmd --fail-fast";;
+			*)
+				remaining="$remaining $*";;
```

Comment:
Not sure you want to use $* here.

`-test-clj -ff -r io.jesi.blah io.jesi.not-blah` becomes
`--fail-fast --watch --focus io.jesi.blah io.jesi.not-blah io.jesi.not-blah`

the loop will collate remaining args as it goes
```suggestion
				remaining="$remaining $1";;
```